src/sources/generator.rs

```diff
@@ -18,25 +18,106 @@ use tokio::time::{interval, Duration};
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct GeneratorConfig {
-    #[serde(default)]
-    sequence: bool,
-    lines: Vec<String>,
     #[serde(default)]
     batch_interval: Option<f64>,
     #[serde(default = "usize::max_value")]
     count: usize,
+    format: OutputFormat,
+}
+
+#[derive(Clone, Debug, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+#[serde(rename_all = "snake_case")]
+pub enum OutputFormat {
+    #[derivative(Default)]
+    RoundRobin {
+        #[serde(default)]
+        sequence: bool,
+        items: Vec<String>,
+    },
+}
+
+impl OutputFormat {
+    async fn generate(
+        &self,
+        config: &GeneratorConfig,
```

Comment:
This may make more sense to also be part of `impl GeneratorConfig` since the `&self` parameter here is tied strictly to the on in the enclosing `config`. Then you would just match on `self.format` below.